Jormungandr: refactor journeys links

antoine-de opened the pull request:
links were done in several places.

they are now done in 1 place: the scenarios

This way each scenario can use the same mechanims as the internal one to fill the journeys with the min_nb_journeys parameter

linked to http://jira.canaltp.fr/browse/NAVITIAII-2079

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/canaltp/navitia/1509)

<!-- Reviewable:end -->

antoine-de committed:
Jormungandr: add a prev/next method in default scenario

antoine-de committed:
Jormungandr: move test method

antoine-de committed:
Jormungandr: emove useless file

antoine-de committed:
Jormungandr: move link creation

to do it in only one place, the link creation is now done in protobuf,
not later in decorator

antoine-de committed:
Jormungandr: plug prev/next in new default

antoine-de committed:
jormungandr: small change to manage different flask version

antoine-de committed:
Jormungandr: force import to load configuration early

TeXitoi commented on source/jormungandr/jormungandr/scenarios/new_default.py:
```diff
@@ -698,3 +671,48 @@ def isochrone(self, request, instance):
+
+        to do that we find the best journey (soonest arrival, then shortest journey) found in the responses
+        and add one second to it's departure.
+        (if anticlockwise, remove one second to the arrival of the journey with latest departure, then shortest)
```
This comment seems deconnected from the code.

kinnou02 commented:
destineo is not handled!

TeXitoi commented on source/jormungandr/jormungandr/scenarios/new_default.py:
```diff
@@ -698,3 +671,48 @@ def isochrone(self, request, instance):
+        #TODO forbid ODTs
+        return request
+
+    def __get_best_for_criteria(self, journeys, criteria):
```
static method?

antoine-de committed:
Jormungandr: destineo prev/next

one BIG down side now is that the destineo journeys test cannot run the
old default test because nosetest run them as 'default' journeys, so the
link check fail (because links are different in 'destineo' and in
'default')

antoine-de commented:
Note: @kinnou02 

one BIG down side now is that the destineo journeys integration tests cannot run the
old default test (thus teh test class cannot inherit from `JourneyTest`) because nosetest run them as 'default' journeys, so the links checks fail (because links are different in 'destineo' and in
'default')

antoine-de committed:
Jormungandr: small test function refacto

pbougue commented on source/jormungandr/tests/routing_tests_experimental.py:
```diff
@@ -62,13 +63,35 @@ def teardown(self):
+    def check_next_datetime_link(dt, response):
+        if not response.get('journeys'):
+            return
+        """default next behaviour is 1 mn after the best or the soonest"""
```
comment is not matching code, 1mn vs 1s

pbougue commented on source/jormungandr/tests/routing_tests_experimental.py:
```diff
@@ -62,13 +63,35 @@ def teardown(self):
+    def check_previous_datetime_link(dt, response):
+        if not response.get('journeys'):
+            return
+        """default previous behaviour is 1 mn before the best or the latest """
```
comment is not matching code, 1mn vs 1s

pbougue commented on source/jormungandr/tests/routing_tests_new_default.py:
```diff
@@ -62,13 +62,38 @@ def teardown(self):
+    def check_next_datetime_link(dt, response):
+        if not response.get('journeys'):
+            return
+        """default next behaviour is 1 mn after the best or the soonest"""
```
comment is not matching code, 1mn vs 1s

pbougue commented on source/jormungandr/tests/routing_tests_new_default.py:
```diff
@@ -62,13 +62,38 @@ def teardown(self):
+    def check_previous_datetime_link(dt, response):
+        if not response.get('journeys'):
+            return
+        """default previous behaviour is 1 mn before the best or the latest """
```
comment is not matching code, 1mn vs 1s

pbougue commented on source/jormungandr/tests/tests_mechanism.py:
```diff
@@ -212,6 +216,104 @@ def query_no_assert(self, url, display=False):
+    def check_next_datetime_link(dt, response):
+        if not response.get('journeys'):
+            return
+        """default next behaviour is 1 mn after the best or the soonest"""
```
minor: 1 `min`

pbougue commented:
Ok for me

antoine-de committed:
minor changes after review

TeXitoi commented:
OK for me.

antoine-de commented:
Jenkins, test this please
